tools/logger/logger_test.go

```diff
@@ -0,0 +1,299 @@
+			},
+			expected: `
+func (s *TimeStruct) MarshalLog(addField func(key string, value interface{})) {
+	if s == nil {
+		return
+	}
+
+addField("created_at", s.CreatedAt.UTC().Format(time.RFC3339Nano))
+}
+`,
+		},
+		{
+			name: "omitempty fields",
+			setup: func() (*types.Struct, string) {
+				fields := []*types.Var{
+					types.NewField(0, nil, "Name", types.Typ[types.String], false),
+					types.NewField(0, nil, "AgeP", types.NewPointer(types.Typ[types.Int]), false),
+				}
+				tags := []string{
+					`log:"name,omitempty"`,
+					`log:"ageP,omitempty"`,
+				}
+				return types.NewStruct(fields, tags), "OmitStruct"
+			},
+			expected: `
+func (s *OmitStruct) MarshalLog(addField func(key string, value interface{})) {
+	if s == nil {
+		return
+	}
+
+if s.Name != "" {
+	addField("name", s.Name)
+}
+if s.AgeP != nil {
+	addField("ageP", s.AgeP)
+}
+}
+`,
+		},
+		{
+			name: "nested marshaler",
+			setup: func() (*types.Struct, string) {
+				nestedType := types.NewPointer(types.NewNamed(
+					types.NewTypeName(0, nil, "NestedStruct", nil),
+					&types.Struct{},
+					nil,
+				))
+				fields := []*types.Var{
+					types.NewField(0, nil, "Nested", nestedType, false),
+				}
+				tags := []string{`log:"."`}
+				return types.NewStruct(fields, tags), "ParentStruct"
+			},
+			expected: `
+func (s *ParentStruct) MarshalLog(addField func(key string, value interface{})) {
+	if s == nil {
+		return
+	}
+
+if s.Nested != nil {
+	s.Nested.MarshalLog(addField)
+}
+}
+`,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s, name := tt.setup()
+			buf := &bytes.Buffer{}
+			processStruct(buf, s, name)
+			assert.Equal(t, tt.expected, buf.String())
+		})
+	}
+}
+
+func TestContains(t *testing.T) {
+	var uninitializedStrings []string
+	var uninitializedIntegers []int
+	var uninitializedBooleans []bool
+	tests := []struct {
+		name           string
+		slice          any
+		item           any
+		underlyingType string
+		expected       bool
+	}{
+		{
+			name:           "string slice with matching item",
+			slice:          []string{"a", "b", "c"},
+			item:           "b",
+			underlyingType: "string",
+			expected:       true,
+		},
+		{
+			name:           "string slice without matching item",
+			slice:          []string{"a", "b", "c"},
+			item:           "d",
+			underlyingType: "string",
+			expected:       false,
+		},
+		{
+			name:           "empty string slice",
+			slice:          []string{},
+			item:           "a",
+			underlyingType: "string",
+			expected:       false,
+		},
+		{
+			name:           "uninitialized string slice",
+			slice:          uninitializedStrings,
+			item:           "a",
+			underlyingType: "string",
+			expected:       false,
+		},
+		{
+			name:           "int slice with matching item",
+			slice:          []int{1, 2, 3},
+			item:           2,
+			underlyingType: "int",
+			expected:       true,
+		},
+		{
+			name:           "uninitialized string slice",
```

Comment:
suggestion: wrong test name?
```suggestion
			name:           "uninitialized int slice",
```

Reply:
Yeah, copy paste.